### PR TITLE
Preparation page for Ubuntu 18.04

### PR DESCRIPTION
**Pull request by mmattel**
Adresses: https://github.com/owncloud/docs/issues/602 (Ubuntu 18.04 php 7.2 ownCloud module dependency fix list)

The intention of this commit is to create a preparation page for Ubuntu 18.04 before ownCloud gets installed. There are too many steps for one page having all OS prepared and the oC installation, so lets split that and get a better overview.

As we know with this page how thinks look like, we can derive relatively easy to other OS and create own documents for them.

It currently does not cover any details to setups after installing like php or redis or mariadb ect.
A second page how to install oC on a prepared server makes a lot of sense and is OS independent.

This page is currently **NOT** linked to the index and is intended for work in progress.
When final, it can be renamed, linked, ect.

You can view the rendered text when clicking "view" in the files changed link above.

### CODE REVIEW

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 1]
```diff
@@ -0,0 +1,199 @@
+= Server Preparation for Ubuntu 18.04
+:hash-installation: http://php.net/manual/en/hash.installation.php
+:mcrypt-link-url: https://websiteforstudents.com/install-php-7-2-mcrypt-module-on-ubuntu-18-04-lts/
+:mcrypt-pecl-url: https://pecl.php.net/package/mcrypt
+:discover-samba-hosts: https://ubuntuforums.org/showthread.php?t=2384959
+:install-mariadb-latest: https://downloads.mariadb.org/mariadb/repositories/#
+
+== Webserver and php
+
+You only need to install __one__ of the following webservers.
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: Why are there two underscores surrounding "one"?

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+ 
+=== Apache Webserver + PHP (supported)
+
+The following command installs Apache webserver and php.
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: Please change the sentence to "The following command installs the Apache web server and PHP."

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+[source,console]
+----
+sudo apt install php libapache2-mod-php apache2
+----
+
+=== Apache Webserver + PHP-FPM (unsupported)
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: Please change all references of "Webserver" to "Web server". I need to create another PR to change other references to it.

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+The following command installs and enables Apache webserver with php-fpm. +
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: Please change this sentence to: "The following command installs and enables the Apache web server with php-fpm."

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+This setup can be used in high thruput environments. For performance
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: s/thruput/throughput/g

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+sudo a2enmod proxy_fcgi
+sudo a2enconf php7.2-fpm
+sudo a2dismod php7.2
+sudo a2dismod mpm_prefork
+sudo a2enmod mpm_event
+sudo a2enmod http2
+sudo service apache2 restart
+----
+
+For examples how to configure this setup see
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: Please change the start of this sentence to "For examples of how to".

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+----
+
+For examples how to configure this setup see 
+http://httpd.apache.org/docs/2.4/mod/mod_proxy.html#handler[link 1] and
+https://httpd.apache.org/docs/2.4/mod/mod_proxy_fcgi.html[link 2] and
+https://wiki.apache.org/httpd/PHP-FPM[link 3]
+
+=== Nginx Webserver + PHP-FPM (unsupported)
+
+The following command installs nginx webserver and php-fpm.
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: Please change this sentence to: "The following command installs the Nginx web server and php-fpm."

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+
+[source,console]
+----
+sudo apt install php-fpm nginx
+----
+
+For details how to configure nginx please see the
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: Please change this sentence to: "For details of how to configure Nginx, please see the"

File: modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+This framework does not need any manual interverntion anymore. +
+
+- As of PHP 5.1.2, the Hash extension is bundled and compiled into PHP by default.
+- As of PHP 7.4.0, the Hash extension is a core PHP extension, so it is always enabled.
+
+For more information see following {hash-installation}[link]
+
+== Exif Library
+
+This library should be already part of php 7.2. +
```
Comment (settermjd) [inline modules/admin_manual/pages/installation/server_prep_ubuntu_18.04.adoc]: Please uppercase "php" in this sentence.

**Comment by mmattel** [comment]
The document itself is now ok, but we need to find a place where we link it into...

**Comment by settermjd** [comment]
How about in `installation/manual_installation.html#on-ubuntu-18.04-lts-server`?

**Comment by settermjd** [comment]
@mmattel, I updated the document across three commits and linked it to the relevant section of the documentation. Can you review the changes and share your feedback, please?

**Comment by mmattel** [comment]
just seen the outcome, will file a commit with additional improvements today.

**Comment by mmattel** [comment]
My last commit (updates and improvements) improves:
- The Ubuntu Preparation page.
I separated the common prerequisites into three blocks for a much better overview.
(Do not concatenate them together, because this was a readbility mess in the other descriptions before)
Plus some improvements added
- The Manual Installation page
By removing double content which is now in the Ubuntu Preparation page.

**Pull request merged**